source/extensions/filters/http/ip_tagging/ip_tagging_filter.cc

```diff
@@ -77,12 +77,19 @@ Http::FilterHeadersStatus IpTaggingFilter::decodeHeaders(Http::RequestHeaderMap&
     return Http::FilterHeadersStatus::Continue;
   }
 
+  if (!config_->customHeader().get().empty()) {
+    headers.remove(config_->customHeader());
+  }
+
   std::vector<std::string> tags =
       config_->trie().getData(callbacks_->streamInfo().downstreamAddressProvider().remoteAddress());
 
   if (!tags.empty()) {
     const std::string tags_join = absl::StrJoin(tags, ",");
     headers.appendEnvoyIpTags(tags_join, ",");
+    if (!config_->customHeader().get().empty()) {
+      headers.setReferenceKey(config_->customHeader(), tags_join);
+    }
```

Comment:
From my perspective, I will prefer to set only one header. The default one, or the custom one.

/wait

Reply:
just to provide a use case of having both the additional header and the original x-envoy-ip-tags header. in cases where people want to migrate from x-envoy-ip-tags to another header, it's beneficial to have the ability to generate both headers, and eventually only tags the new header. So it'd be helpful if we can tag both headers for migration.

Reply:
The behavior of new field should be documented clearly in the API. And after it's documented, we cannot change it easily. It's not a good choice to compromise for un-certain `migration` (And I don't get why a user will wont to migrate to new header if the previous header work well.)

And If there actually is a similar requirement/scenario (two headers are necessary), it could be achieved by setting two ip tag filters in the `http_filters`.